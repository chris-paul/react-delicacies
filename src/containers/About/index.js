/*
 * @Author: your name
 * @Date: 2019-12-28 10:16:47
 * @LastEditTime: 2021-05-15 12:13:16
 * @LastEditors: 廉恒凯
 * @Description: In User Settings Edit
 * @FilePath: \react-delicacies\src\pages\Login\index.js
 */

import React from 'react';
import { Avatar, List } from 'antd';
import styles from './index.less';

const data = [
    {
        author: 'chris-paul',
        avatar:
            'https://avatars.githubusercontent.com/u/18257755?s=400&u=20fdd5ea3dd65ebcbc90492130871acbfcab4222&v=4',
        content: (
            <p>
                现就职于上海某公司担任前端开发工程师一职,热爱前端技术,是一名狂热的足球、篮球爱好者,个人邮箱chrispaullhk@gmail.com,欢迎骚扰。
            </p>
        ),
    },
    {
        author: 'MMLucy',
        avatar: 'https://avatars.githubusercontent.com/u/23635143?v=4',
        content: <p>上海某公司前端开发工程师,重庆妹子一枚,酷爱旅游,欢迎大家踊跃star。</p>,
    },
];

const About = () => (
    <List
        className={styles.commentList}
        header="关于我们"
        data-test="about"
        itemLayout="horizontal"
        dataSource={data}
        renderItem={item => (
            <List.Item>
                <List.Item.Meta
                    title={item.author}
                    avatar={<Avatar src={item.avatar} />}
                    description={item.content}
                />
            </List.Item>
        )}
    />
);

export default About;
