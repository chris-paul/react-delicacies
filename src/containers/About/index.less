
.commentList {
    width: 70%;
    margin: 30px auto;
}
