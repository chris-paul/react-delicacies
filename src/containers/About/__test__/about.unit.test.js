/*
 * @Author:廉恒凯
 * @Date: 2020-01-05 19:41:10
 * @LastEditTime: 2021-05-15 14:35:57
 * @LastEditors: 廉恒凯
 * @Description: In User Settings Edit
 * @FilePath: /react-delicacies/src/pages/Work/__tests__/spec.js
 */
import React from 'react';
import { shallow } from 'enzyme';
import About from '../index';

const findWrapper = (wrapper, tag) => {
    return wrapper.find(`[data-test="${tag}"]`);
};

describe('About', () => {
    it('should show despriction success', () => {
        const wraper = shallow(<About />);
        const ListElem = findWrapper(wraper, 'about');
        expect(ListElem.length).toBe(1);
        expect(wraper).toMatchSnapshot();
    });
});
