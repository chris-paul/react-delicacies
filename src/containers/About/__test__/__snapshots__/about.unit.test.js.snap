// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should show despriction success 1`] = `
<List
  data-test="about"
  dataSource={
    Array [
      Object {
        "author": "chris-paul",
        "avatar": "https://avatars.githubusercontent.com/u/18257755?s=400&u=20fdd5ea3dd65ebcbc90492130871acbfcab4222&v=4",
        "content": <p>
          现就职于上海某公司担任前端开发工程师一职,热爱前端技术,是一名狂热的足球、篮球爱好者,个人邮箱chrispaullhk@gmail.com,欢迎骚扰。
        </p>,
      },
      Object {
        "author": "MMLucy",
        "avatar": "https://avatars.githubusercontent.com/u/23635143?v=4",
        "content": <p>
          上海i 云保公司前端开发工程师,重庆妹子一枚,酷爱旅游,欢迎大家踊跃star。
        </p>,
      },
    ]
  }
  header="关于我们"
  itemLayout="horizontal"
  renderItem={[Function]}
/>
`;
